File: data/readable_format/fortran/file162.f90
module volmod
  implicit none
  integer, volatile :: a
  logical, volatile :: b
  logical :: c
contains
  subroutine sample
    a = 33
    if (a /= 432) then
      print *, 'aPresent'
    endif

    b = .false.
    if (b) then
      print *, 'bPresent'
    endif

    c = .false.
    if (c) then
      print *, 'cPresent'
    endif
  end subroutine sample
end module volmod

program main
  use volmod
  implicit none

  ! Test setup
  a = 432
  b = .false.
  c = .false.

  ! Initial state checks
  if (a == 432) then
    print *, 'Test 1a: a should initially be 432 - PASSED'
  else
    print *, 'Test 1a: a should initially be 432 - FAILED'
  endif

  if (.not. b) then
    print *, 'Test 1b: b should initially be .false. - PASSED'
  else
    print *, 'Test 1b: b should initially be .false. - FAILED'
  endif

  if (.not. c) then
    print *, 'Test 1c: c should initially be .false. - PASSED'
  else
    print *, 'Test 1c: c should initially be .false. - FAILED'
  endif

  ! Call the subroutine and check effects
  call sample

  if (a == 33) then
    print *, 'Test 2a: a should be 33 after sample - PASSED'
  else
    print *, 'Test 2a: a should be 33 after sample - FAILED'
  endif

  if (.not. b) then
    print *, 'Test 2b: b should remain .false. after sample - PASSED'
  else
    print *, 'Test 2b: b should remain .false. after sample - FAILED'
  endif

  if (.not. c) then
    print *, 'Test 2c: c should remain .false. after sample - PASSED'
  else
    print *, 'Test 2c: c should remain .false. after sample - FAILED'
  endif
end program main

File: data/readable_format/cpp/file162.cpp
#include <iostream>
#include <cassert> // Include the cassert header for assert()

class volmod {
public:
    static volatile int a;
    static volatile bool b; // Made volatile as per Fortran code
    static bool c;

    static void sample() {
        a = 33;
        if(a != 432) std::cout << "aPresent" << std::endl;

        b = false;
        if(b) std::cout << "bPresent" << std::endl;

        c = false;
        if(c) std::cout << "cPresent" << std::endl;
    }
};

// Define static members outside the class
volatile int volmod::a;
volatile bool volmod::b;
bool volmod::c;

void test_volmod() {
    // Initial state checks
    volmod::a = 432;
    assert(volmod::a == 432);
    volmod::b = false;
    assert(!volmod::b);
    volmod::c = false;
    assert(!volmod::c);

    // Call the sample and check effects
    volmod::sample();
    assert(volmod::a == 33);
    assert(!volmod::b);
    assert(!volmod::c);

    std::cout << "All tests passed." << std::endl;
}

int main() {
    test_volmod();
    return 0;
}